presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2541,6 +2542,102 @@ public void testPredicatePushDownToTableScan()
         }
     }
 
+    @Test
+    public void testPartitionPruning()
+    {
+        assertUpdate("CREATE TABLE test_partition_pruning (v bigint, k varchar) WITH (partitioned_by = array['k'])");
+        assertUpdate("INSERT INTO test_partition_pruning (v, k) VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'e')", 4);
```

Comment:
I need the type of the field to be different from the value that gets used in the query so that the planner inserts some implicit coercions. Otherwise, the plan ends up with the "right enough" shape to not trigger the problem. I can make it a CTAS with casts or explicit value constructors, but I'll also add a comment, since this is a non-obvious detail.